### PR TITLE
Move docker state tests to pytest

### PR DESCRIPTION
**Pull request by krionbsd**
### What does this PR do?
Move docker state tests to pytest

### CODE REVIEW

File: tests/pytests/unit/states/test_docker_image.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        "docker.pull": docker_pull,
+        "docker.inspect_image": docker_inspect_image,
+        "docker.resolve_tag": docker_resolve_tag,
+    }
+    with patch.dict(docker_state.__dict__, {"__salt__": __salt__}):
+        ret = docker_state.present("image:latest", force=True)
+        assert ret == {
+            "changes": {},
+            "result": True,
+            "comment": "Image 'image:latest' was pulled, " "but there were no changes",
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_docker_image.py:53]:
```suggestion
            "comment": "Image 'image:latest' was pulled, but there were no changes",
```

File: tests/pytests/unit/states/test_docker_volume.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        assert {
+            "name": "volume_foo",
+            "comment": (
+                "Driver for existing volume 'volume_foo'"
+                " ('dummy_default') does not match specified"
+                " driver ('local') and force is False"
+            ),
+            "changes": {},
+            "result": False,
+        } == ret
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_docker_volume.py:84]:
```suggestion
        assert ret == {
            "name": "volume_foo",
            "comment": (
                "Driver for existing volume 'volume_foo'"
                " ('dummy_default') does not match specified"
                " driver ('local') and force is False"
            ),
            "changes": {},
            "result": False,
        }
```
this makes more sense on failures when pytest shows us a diff, it should always be `what_we_got == what_we_expect`

File: tests/pytests/unit/states/test_docker_volume.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        assert orig_volumes == volumes
+
+        # run it again with a different driver and force
+        ret = docker_state.present("volume_foo", driver="local", force=True)
+        assert {
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_docker_volume.py:89]: same as above

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/states/test_docker_image.py (hunk 1 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_docker_image.py]: Fixed

File: tests/pytests/unit/states/test_docker_volume.py (hunk 2 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_docker_volume.py]: yeah right, I missed it somehow, fixed now

File: tests/pytests/unit/states/test_docker_volume.py (hunk 3 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_docker_volume.py]: Fixed

**Review by twangboy — APPROVED** [review]

**Review by twangboy — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**